tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Item Update Bulk Feature.feature

```diff
@@ -0,0 +1,111 @@
+﻿Feature: Update order items (bulk)
+    As a buyer user
+    I want to update order items for a given order
+    So that I can complete my order
+
+Background:
+    Given Orders exist
+        | OrderId    | Description      | LastUpdatedBy                        | LastUpdatedByName | OrganisationId                       | CommencementDate |
+        | C000014-01 | Some Description | 335392e4-4bb1-413b-9de5-36a85c9c0422 | Tom Smith         | 4af62b99-638c-4247-875e-965239cd0c48 | 01/01/2021       |
+    And Service Recipients exist
+        | OdsCode | Name        | OrderId    |
+        | ODS1    | Recipient 1 | C000014-01 |
+        | ODS2    | Recipient 2 | C000014-01 |
+        | ODS3    | Recipient 3 | C000014-01 |
+    And Order items exist
+        | OrderId    | OdsCode | CatalogueItemName | CatalogueItemType | ProvisioningType |
+        | C000014-01 | ODS1    | Item A            | Solution          | OnDemand         |
+        | C000014-01 | ODS2    | Item B            | AdditionalService | Declarative      |
+        | C000014-01 | ODS3    | Item C            | AssociatedService | Patient          |
+    And the user is logged in with the Buyer role for organisation 4af62b99-638c-4247-875e-965239cd0c48
+
+@10516
+Scenario: Edit all order items
+    Given the user creates a request to update the order with ID 'C000014-01' with the following items
+        | OrderItemId | ItemType           | PayloadType    | CatalogueItemName | OdsCode | ServiceRecipientName |
+        | 1           | catalogue solution | complete       | Item A            | ODS1    | Recipient 1          |
+        | 2           | additional service | high-boundary  | Item B            | ODS2    | Recipient 2          |
+        | 3           | associated service | low-boundary   | Item C            | ODS3    | Recipient 3          |
+    When the user sends the edit order items request
+    Then a response with status code 204 is returned
+    And the expected order items exist
+    And the persisted service recipients are
+        | OrderId    | OdsCode | Name        |
+        | C000014-01 | ODS1    | Recipient 1 |
+        | C000014-01 | ODS2    | Recipient 2 |
+
+@10516
+Scenario: Add an order item
+    Given the user creates a request to update the order with ID 'C000014-01' with the following items
+        | OrderItemId | ItemType           | PayloadType | CatalogueItemName | OdsCode | ServiceRecipientName |
+        | 1           | catalogue solution | complete    | Item A            | ODS1    | Recipient 1          |
+        | 2           | additional service | complete    | Item B            | ODS2    | Recipient 2          |
+        | 3           | associated service | complete    | Item C            | ODS3    | Recipient 3          |
+        |             | catalogue solution | complete    | Item D            | ODS1    | Recipient 1          |
```

Comment:
Empty OrderItemId?

Reply:
Yeah, that's correct here because I'm testing the addition of a new item to the order. A new order item won't have an order item ID. If an order item ID is provided then it's treated as an update, assuming an order item exists with that ID – if not then it's a validation error.

Reply:
righto, I'll approve the PR